CMakeLists.txt

```diff
@@ -25,8 +25,6 @@ find_package(rclcpp REQUIRED)
 find_package(Eigen3 REQUIRED)
 find_package(GDAL 3 REQUIRED)
 
-find_package(yaml_cpp_vendor REQUIRED)
```

Comment:
Need to revert this as yaml-cpp does not resolve correctly on macOS. Suggest:

```c++
# Fix for yaml_cpp not resolving correctly on macOS
if (APPLE)
  find_package(yaml_cpp_vendor REQUIRED)
  link_directories(${yaml_cpp_vendor_LIBRARY_DIRS})
endif()
```

to make it clear why it's included.

Reply:
@srmainwaring Thanks for pointing this out. Probably means we need a macOS CI for this package

Reply:
Yes, though the problem is that the pre-requisite is a functioning humble build (for CI) and that's expensive as needs to be built from source. Ideally there would be a macOS runner with ROS 2 support we could pull down, put I don't think this exists.

Reply:
I didn't see yaml being used which is the reason I took it out; can you share the compile or runtime error you have if this is absent?

Reply:
You won't see in on Ubuntu. The compiler error is an unresolved library (yaml-cpp) on macOS. This is because the yaml_cpp_vendored library (transitive) dependencies do not seem to work correctly on macOS. It's a pain, should be fixed upstream, but macOS is Tier 3 support so is unlikely to happen.

Reply:
Should we create an upstream issue just to track?
https://github.com/ros2/yaml_cpp_vendor/issues

Reply:
Patch: https://github.com/ethz-asl/grid_map_geo/pull/60